### PR TITLE
feat: introduce history events

### PR DESCRIPTION
**Pull request by Norlock**
# Description of change
Please write a summary of your changes and why you made them.

## Links to any relevant issues
fixes issue #62.
Is based upon: https://github.com/impierce/identity-wallet/pull/122

## How the change has been tested
I ran the Shenron profile and made sure the events where added in the general profile and connection specific.

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/format-lint-test.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     - name: Use Node.js
       uses: actions/setup-node@v4
       with:
-        node-version: '20.x'
+        node-version: '21.x'
```
Comment (daniel-mader) [inline .github/workflows/format-lint-test.yaml]: Any specific reason for this change? `20.x` is the current LTS, no?

File: unime/src/lib/components/molecules/ListItemCard.svelte [hunk 2]
```diff
@@ -10,6 +10,15 @@
   export let description: string | undefined = undefined;
   export let type: 'data' | 'badge' = 'data';
   export let isTempAsset = false;
+  export let hasBorderRadius = true;
+
+  function addRoundedBorder() {
```
Comment (daniel-mader) [inline unime/src/lib/components/molecules/ListItemCard.svelte]: Please use `twMerge()` and `$$props.class` to extend a component. An example can be found in `ButtonRounded.svelte`.

File: unime/src-tauri/tests/fixtures/states/credential_redirect_me.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "credentials": [
         {
             "id": "39373933-3863-3339-3864-646234373631",
-            "issuer_name": null,
+            "issuer_name": "",
```
Comment (daniel-mader) [inline unime/src-tauri/tests/fixtures/states/credential_redirect_me.json:13]: Is an empty string really better than `null`? Null can be more easily checked against and an empty string would be valid, but not display anything. Is this desired behavior?

File: unime/src/i18n/de-DE/index.ts [hunk 4]
```diff
@@ -255,6 +255,9 @@ const de = {
   },
   TIMELINE: {
     EMPTY: 'Noch keine Aktivitäten.',
+    ADDED_CREDENTIALS: 'TODO',
```
Comment (daniel-mader) [inline unime/src/i18n/de-DE/index.ts]: Let's not do this as such things are easily forgotten compared to a type error when using the **_typesafe_**-i18n library. Touching a translation file without adding the actual translations should be avoided.

File: unime/src/i18n/i18n-types.ts [hunk 5]
```diff
@@ -1463,6 +1475,18 @@ export type TranslationFunctions = {
 		 * No activity yet.
 		 */
 		EMPTY: () => LocalizedString
+		/**
+		 * Received credentials from
+		 */
+		ADDED_CREDENTIALS: () => LocalizedString
```
Comment (daniel-mader) [inline unime/src/i18n/i18n-types.ts]: Have you also translated these new types to `en` and `nl-NL`?

File: identity-wallet/src/state/reducers/history.rs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    pub issuer_name: String,
+    pub event_type: EventType,
+    pub connection_id: Option<String>,
+    pub date: String,
+    pub credentials: Vec<HistoryCredential>,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub struct HistoryCredential {
```
Comment (daniel-mader) [inline identity-wallet/src/state/reducers/history.rs]: Do you think it makes sense to add the credential id as field so we could directly jump to that credential when scrolling through the history?

File: identity-wallet/src/command.rs [hunk 7]
```diff
@@ -102,6 +102,9 @@ pub async fn handle_action<R: tauri::Runtime>(
 pub fn emit_event<R: tauri::Runtime>(window: &tauri::Window<R>, app_state: &AppState) -> anyhow::Result<()> {
     const STATE_CHANGED_EVENT: &str = "state-changed";
     window.emit(STATE_CHANGED_EVENT, app_state)?;
-    debug!("emitted event `{}` with payload `{:?}`", STATE_CHANGED_EVENT, app_state);
+
+    let json_str = serde_json::to_string(app_state).unwrap();
```
Comment (daniel-mader) [inline identity-wallet/src/command.rs]: I suggest either inlining into `debug!()` or give it a proper name such as `app_state_json_string`.

File: identity-wallet/src/state/reducers/history.rs [hunk 8]
```diff
@@ -0,0 +1,31 @@
+use serde::{Deserialize, Serialize};
+use ts_rs::TS;
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub enum EventType {
+    #[serde(rename = "added_connection")]
+    AddedConnection,
```
Comment (daniel-mader) [inline identity-wallet/src/state/reducers/history.rs]: The [naming convention for events](https://www.eventstore.com/blog/whats-in-an-event-name#naming-conventions) is **SubjectObjectPredicate** so: `AddedConnection` --> `ConnectionAdded`

File: .github/workflows/format-lint-test.yaml (hunk 1 above)
Comment (Norlock) [reply .github/workflows/format-lint-test.yaml]: I had problems with format not working locally so I changed this one, (eventually it was tsconfig.json that was formatted incorrectly so I added it to the prettierignore). I can revert this change I thought it was fine because it still worked and was newer

File: unime/src-tauri/tests/fixtures/states/credential_redirect_me.json (hunk 3 above)
Comment (Norlock) [reply unime/src-tauri/tests/fixtures/states/credential_redirect_me.json:13]: Yes I also discussed with @nanderstabel  that issuer_name is always set in the code, so it can never be null anyway. We need to make a new PR in the future where we look at the frontend ?? checks on the issuer_name to incorporate those back to the backend again.

File: identity-wallet/src/state/reducers/history.rs (hunk 6 above)
Comment (Norlock) [reply identity-wallet/src/state/reducers/history.rs]: Hmm yes seems like a good idea, I will add the field to it.

File: identity-wallet/src/state/reducers/history.rs (hunk 8 above)
Comment (Norlock) [reply identity-wallet/src/state/reducers/history.rs]: Ah ok I will rename :+1:

File: unime/src/i18n/de-DE/index.ts (hunk 4 above)
Comment (Norlock) [reply unime/src/i18n/de-DE/index.ts]: ok no problem, I do think we should aim for strict typescript compiling, which means that no errors can exist in typing. Which would make this solution not possible, but for now I remove the fields again

File: unime/src/lib/components/molecules/ListItemCard.svelte (hunk 2 above)
Comment (Norlock) [reply unime/src/lib/components/molecules/ListItemCard.svelte:15]: But I want to always use the same class, except for in history I don't want to add a certain class. Isn't it better to do it like this? Otherwise everywhere you have to pass rounded-lg class

File: .github/workflows/format-lint-test.yaml (hunk 1 above)
Comment (Norlock) [reply .github/workflows/format-lint-test.yaml]: changed it back to 20.x

File: unime/src/i18n/de-DE/index.ts (hunk 4 above)
Comment (daniel-mader) [reply unime/src/i18n/de-DE/index.ts]: Yes, but throwing type errors when translations are incomplete is the main idea behind **_typesafe_**-i18n. Keeping all translations up-to-date at all time for all features is a challenge. Maybe we find an elegant solution when it's the right time. 👍

File: identity-wallet/src/state/reducers/user_data_query.rs
Comment (daniel-mader) [inline identity-wallet/src/state/reducers/user_data_query.rs:1]: I'm not questioning these improvements, but is it really necessary to clean up `user_data_query.rs` in the same PR that introduces history events? I know it's just "small issues", but they still delay the review process and bloat the initial scope.

File: identity-wallet/src/state/user_prompt.rs [hunk 9]
```diff
@@ -33,9 +33,9 @@ pub enum CurrentUserPrompt {
         issuer_name: String,
         #[ts(optional)]
         logo_uri: Option<String>,
-        #[ts(type = "object")]
+        #[ts(type = "any")]
```
Comment (daniel-mader) [inline identity-wallet/src/state/user_prompt.rs:36]: Is this related to history events?

File: identity-wallet/src/verifiable_credential_record.rs [hunk 10]
```diff
@@ -68,18 +74,40 @@ impl From<CredentialFormats<WithCredential>> for VerifiableCredentialRecord {
...
+        Value::Array(array) => array.last()?.as_str().map(|name| name.to_string()),
+        _ => None,
+    }
+}
+
 #[derive(Debug, Serialize, Deserialize, Clone, Eq, Derivative, TS)]
 #[derivative(PartialEq)]
 #[ts(export, export_to = "bindings/display-credential/DisplayCredential.ts")]
 pub struct DisplayCredential {
+    /// Also display_icon
```
Comment (daniel-mader) [inline identity-wallet/src/verifiable_credential_record.rs]: I don't think we need to comment that the `id` is also used somewhere else as it's not relevant for the `DisplayCredential`. Also, it's not "display_icon", rather "image id".

File: unime/src/lib/credentials/CredentialList.svelte [hunk 11]
```diff
@@ -98,9 +98,7 @@
     {#each credentials as credential}
       <ListItemCard
         id={credential.id}
-        title={credential.metadata.display.name ??
-          credential.data.credentialSubject.achievement?.name ??
-          credential.data.type.at(-1)}
+        title={credential.display_name}
```
Comment (daniel-mader) [inline unime/src/lib/credentials/CredentialList.svelte:101]: Are we still making sure that there is _something_ displayed in any case? If `display_name` can be an empty string, then we won't see anything in the UI, right?

File: unime/src/routes/(app)/activity/connection/[id]/+page@.svelte [hunk 12]
```diff
@@ -30,9 +27,9 @@
         <ConnectionData id={connection.id} />
       </div>
 
-      <div slot="2" class="h-full bg-silver py-5 dark:bg-navy">
+      <div slot="2" class="bg-silver pt-5 dark:bg-navy">
```
Comment (daniel-mader) [inline unime/src/routes/(app)/activity/connection/[id]/+page@.svelte:30]: Why remove the padding from the bottom?

File: unime/src/routes/(app)/activity/+page.svelte [hunk 13]
```diff
@@ -24,7 +24,7 @@
       </button>
     </div> -->
   </div>
-  <div class="flex grow flex-col overflow-y-auto px-4 py-5">
+  <div class="flex grow flex-col overflow-y-auto px-4 pt-5">
```
Comment (daniel-mader) [inline unime/src/routes/(app)/activity/+page.svelte:27]: Why remove the padding from the bottom?

Comment (daniel-mader) [reply unime/src/routes/(app)/activity/+page.svelte:27]: Okay, overflows are looking nicer 👍

File: unime/.prettierignore [hunk 14]
```diff
@@ -5,6 +5,7 @@ node_modules
 /package
 .env
 .env.*
+.tsconfig.json
```
Comment (daniel-mader) [inline unime/.prettierignore]: Did you have problems with `tsconfig` being formatted differently on your machine?

File: identity-wallet/src/state/reducers/history.rs [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+    pub date: String,
+    pub credentials: Vec<HistoryCredential>,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub struct HistoryCredential {
+    pub title: String,
+    pub sub_title: String,
+    /// Is also the image id
```
Comment (daniel-mader) [inline identity-wallet/src/state/reducers/history.rs]: I don't think we need to be aware here that this `id` is used elsewhere. You can safely remove this comment.

File: identity-wallet/src/state/reducers/history.rs [hunk 16]
```diff
@@ -0,0 +1,32 @@
+use serde::{Deserialize, Serialize};
+use ts_rs::TS;
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub enum EventType {
+    #[serde(rename = "connection_added")]
```
Comment (daniel-mader) [inline identity-wallet/src/state/reducers/history.rs]: Why rename? `"ConnectionAdded"` is perfectly idiomatic for the frontend as well.

File: identity-wallet/bindings/EventType.ts [hunk 17]
```diff
@@ -0,0 +1,3 @@
+// This file was generated by [ts-rs](https://github.com/Aleph-Alpha/ts-rs). Do not edit this file manually.
+
+export type EventType = "connection_added" | "credentials_added" | "credentials_shared";
```
Comment (daniel-mader) [inline identity-wallet/bindings/EventType.ts]: I think `"ConnectionAdded"` is more conventional than `"connection_added"` for the frontend.

File: identity-wallet/src/state/user_prompt.rs (hunk 9 above)
Comment (Norlock) [reply identity-wallet/src/state/user_prompt.rs:36]: Its part of the other PR where this one was built upon.

File: identity-wallet/src/state/reducers/user_data_query.rs
Comment (Norlock) [reply identity-wallet/src/state/reducers/user_data_query.rs:1]: It was needed because display_name has been set in the backend and is not an optional value anymore. ALso part of the previous PR

File: identity-wallet/src/verifiable_credential_record.rs (hunk 10 above)
Comment (Norlock) [reply identity-wallet/src/verifiable_credential_record.rs]: Well it just makes everything easier. Because last time we took a long time finding out where to find the "image id" and we had to look through the code until you realized it was that field.

File: unime/.prettierignore (hunk 14 above)
Comment (Norlock) [reply unime/.prettierignore]: Yes it adds trailing slashes on that file, which will fail again on github actions

File: unime/src/lib/credentials/CredentialList.svelte (hunk 11 above)
Comment (Norlock) [reply unime/src/lib/credentials/CredentialList.svelte:101]: Yes because this is being added in the backend. Its part of the previous PR, we talked about few weeks ago where I said I would move this to the backend.

File: identity-wallet/src/verifiable_credential_record.rs (hunk 10 above)
Comment (daniel-mader) [reply identity-wallet/src/verifiable_credential_record.rs]: Let's make a proper comment out of it then such as "This id is also used as the image asset id"

File: unime/src/lib/credentials/CredentialList.svelte (hunk 11 above)
Comment (daniel-mader) [reply unime/src/lib/credentials/CredentialList.svelte:101]: Maybe I commented in the wrong place (frontend), but would we handle `display_name` to never be an empty string?

Comment (Norlock) [reply unime/src/lib/credentials/CredentialList.svelte:101]: Well it can be an empty string but the whole idea of a display_name is that it's always set right? I think we should require that some fields are always set. Because to have a name missing of a credential in my opinion is not a correct state. Because to the end user it's hard to make sense of what the credential is otherwise right

Comment (daniel-mader) [reply unime/src/lib/credentials/CredentialList.svelte:101]: Yes, correct. That's what I was asking. We should make sure it can never be empty in the backend. We do that, right?

Comment (Norlock) [reply unime/src/lib/credentials/CredentialList.svelte:101]: I just moved those three ?? fields to the backend. If something is still missing it would have been null or empty before as well, so maybe for another story. To see where its added now: identity-wallet/src/verifiable_credential_record.rs

**Comment by nanderstabel** [comment]
Currently when I connect with the same Connection multiple times, they will all show up as a "First connection". I believe from the 
#62 after the first connection we should refer to 're-connecting' as simply "Connected to ...".

Screenshot_20240229-175705

File: unime/src/lib/events/History.svelte [hunk 18]
```diff
@@ -1,71 +1,99 @@
...
+  import type { HistoryEvent } from '@bindings/HistoryEvent';
+
   import HistoryEntry from '$lib/events/HistoryEntry.svelte';
   import LL from '$src/i18n/i18n-svelte';
-  import exampleEvents from '$src/lib/events/mock-data.json';
   import { state } from '$src/stores';
 
-  // const events: Event[] = exampleEvents.map((e) => ({ ...e, type: e.type as EventType }));
-  const events: Event[] = [];
+  import DownloadSimple from '~icons/ph/download-simple';
```
Comment (daniel-mader) [inline unime/src/lib/events/History.svelte]: I think we're using the "fill" version so you could use:

```
import DownloadSimple from '~icons/ph/download-simple-fill';
import PlugsConnected from '~icons/ph/plugs-connected-fill';
import ShareFat from '~icons/ph/share-fat-fill';
```

File: unime/src/lib/events/History.svelte [hunk 19]
```diff
@@ -1,71 +1,99 @@
...
+              <svelte:component this={event.icon} class="h-4 w-4 " />
+            </div>
+            {#if hasNextElement(i)}
+              <!-- Vertical line on the left -->
+              <div class="mb-2 mt-4 h-full rounded-full border border-slate-200 border-y-gray-200"></div>
+            {/if}
           </div>
-          <div class="grow">
-            <HistoryEntry {...event} />
+          <div class="ml-6 mt-[-8px] flex grow justify-between pb-10">
```
Comment (daniel-mader) [inline unime/src/lib/events/History.svelte]: If you want to align it in the middle, you'd need `mt-[-5px]` since two lines of text are `42px` and the icon to left is `32px`, so `(42px - 32px) / 2 = 5px`.

Comment (daniel-mader) [reply unime/src/lib/events/History.svelte]: Using a non-absolute way of alignment would be better (using flex), but it's okay for now.

File: unime/src/lib/events/History.svelte [hunk 20]
```diff
@@ -1,71 +1,99 @@
...
+    <div class="ml-2 mt-6">
+      {#each events as event, i}
+        <div class="flex flex-row">
+          <div class="mt-2 flex flex-col items-center">
+            <div class="z-10 flex items-center justify-center rounded-full bg-white ring-8 ring-white dark:ring-silver">
+              <svelte:component this={event.icon} class="h-4 w-4 " />
+            </div>
+            {#if hasNextElement(i)}
+              <!-- Vertical line on the left -->
+              <div class="mb-2 mt-4 h-full rounded-full border border-slate-200 border-y-gray-200"></div>
```
Comment (daniel-mader) [inline unime/src/lib/events/History.svelte]: Using borders here is not necessary, you can just use a regular width, such as `w-[2px]` and remove all borders. Also, please add `bg-grey dark:bg-blue`.

File: unime/src/lib/events/History.svelte [hunk 21]
```diff
@@ -1,71 +1,99 @@
...
-    <div class="flex grow flex-col space-y-8 pr-4 pt-4">
-      {#each eventsList as event}
-        <div class="flex justify-between">
-          <div class="z-10 mr-3 h-6 w-6 overflow-hidden rounded-full bg-white p-0.5 ring-8 ring-silver">
-            <Image id={event.connection.id} imgClass="h-full w-full" />
+    <div class="ml-2 mt-6">
+      {#each events as event, i}
+        <div class="flex flex-row">
+          <div class="mt-2 flex flex-col items-center">
+            <div class="z-10 flex items-center justify-center rounded-full bg-white ring-8 ring-white dark:ring-silver">
```
Comment (daniel-mader) [inline unime/src/lib/events/History.svelte]:
```suggestion
            <div class="z-10 flex items-center justify-center rounded-full bg-grey text-slate-800 ring-8 ring-grey dark:bg-blue dark:text-grey dark:ring-blue">
```

Comment (daniel-mader) [reply unime/src/lib/events/History.svelte]: There colors are a little bit off here in comparison with the UI designs.

File: identity-wallet/src/state/reducers/authorization.rs [hunk 22]
```diff
@@ -261,19 +279,33 @@ pub async fn handle_oid4vp_authorization_request(state: AppState, action: Action
...
+
         let verifiable_credentials: Vec<Credential<JwtVcJson>> = stronghold_manager
             .values()
             .map_err(StrongholdValuesError)?
             .unwrap()
             .iter()
-            .filter_map(
-                |verifiable_credential_record| match &verifiable_credential_record.verifiable_credential {
+            .filter_map(|verifiable_credential_record| {
+                let share_cred = match &verifiable_credential_record.verifiable_credential {
```
Comment (nanderstabel) [inline identity-wallet/src/state/reducers/authorization.rs]: Preferably please follow the general naming convention by avoiding abbreviations :)

File: identity-wallet/src/state/reducers/credential_offer.rs [hunk 23]
```diff
@@ -359,25 +366,40 @@ pub async fn send_credential_request(state: AppState, action: Action) -> Result<
...
             .map(|verifiable_credential_record| verifiable_credential_record.display_credential)
             .collect();
 
-        let current_user_prompt = CurrentUserPrompt::Redirect {
-            target: "me".to_string(),
-        };
+        // History
+        if !history_credentials.is_empty() {
+            state.history.push(HistoryEvent {
+                issuer_name: issuer_name.clone(),
```
Comment (nanderstabel) [inline identity-wallet/src/state/reducers/credential_offer.rs]: This does not need to be cloned :+1:

File: unime/src/lib/events/HistoryEntry.svelte [hunk 24]
```diff
@@ -1,34 +1,75 @@
...
         <ListItemCard
           id={credential.id}
-          title={credential.metadata.display.name ?? credential.data.type.at(-1)}
-          description="Lorem ipsum dolor"
-        >
-          <!-- <span slot="icon"><EnvelopeSimple class="h-6 w-6" /></span> -->
-        </ListItemCard>
+          title={credential.title}
+          description={credential.sub_title}
+          class="rounded-none"
```
Comment (daniel-mader) [inline unime/src/lib/events/HistoryEntry.svelte]: Why remove the rounded borders?
* It does not have a visual effect since background and element have the same color.
* Also when they don't have the same color, it looks worse without rounded borders.

File: unime/src/lib/events/HistoryEntry.svelte [hunk 25]
```diff
@@ -1,34 +1,75 @@
...
-      dateStyle: 'medium',
-      timeStyle: 'medium',
-    })}
+  <p class="text-[14px]/[22px] font-medium text-slate-800 dark:text-grey">{title}</p>
+  <p class="text-[12px]/[20px] font-medium text-slate-500 dark:text-slate-300">
+    {getRelativeDate()}
   </p>
   {#if credentials.length > 0}
-    <div class="mt-[12px] rounded-xl border border-slate-200 bg-white p-[3px] dark:border-slate-600">
+    <div class="mt-[12px] rounded-xl border border-y-gray-200 bg-white p-[3px] dark:border-dark dark:bg-dark">
```
Comment (daniel-mader) [inline unime/src/lib/events/HistoryEntry.svelte]: * These changes do not reflect the UI designs (they use `slate`, not `gray`).
* Why are only the y-axis borders styled?

File: identity-wallet/bindings/HistoryCredential.ts [hunk 26]
```diff
@@ -0,0 +1,3 @@
+// This file was generated by [ts-rs](https://github.com/Aleph-Alpha/ts-rs). Do not edit this file manually.
+
+export interface HistoryCredential { title: string, sub_title: string, id: string, }
```
Comment (daniel-mader) [inline identity-wallet/bindings/HistoryCredential.ts]: `sub_title` is non-conventional naming for JS/TS. It always contains the `issuerName`, right? So you could also just use that. If you want to keep it more generic, you could also name it `description` (as in the generic `ListItemCard`).

File: unime/src/lib/events/History.svelte (hunk 21 above)
Comment (Norlock) [reply unime/src/lib/events/History.svelte]: I only see light theme design? Where did you see the dark theme design?

File: identity-wallet/src/state/reducers/history.rs [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+pub enum EventType {
+    ConnectionAdded,
+    CredentialsAdded,
+    CredentialsShared,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub struct HistoryEvent {
+    pub issuer_name: String,
```
Comment (nanderstabel) [inline identity-wallet/src/state/reducers/history.rs]: Probably better to change this to a more generic name, such as `name` or `connection_name`. 

In most cases issuers and verifiers are very distinct entities so `issuer_name` is not the right term when credentials are shared.

File: unime/src/lib/events/History.svelte (hunk 21 above)
Comment (daniel-mader) [reply unime/src/lib/events/History.svelte]: All features are designed only in light mode and we have to apply the dark theme ourselves by following the color mapping.

File: unime/src/i18n/en/index.ts [hunk 28]
```diff
@@ -254,6 +254,9 @@ const en = {
   },
   TIMELINE: {
     EMPTY: 'No activity yet.',
+    CREDENTIALS_ADDED: 'Received credentials from',
+    CREDENTIALS_SHARED: 'Shared credentials with',
```
Comment (daniel-mader) [inline unime/src/i18n/en/index.ts]: Isn't it "Shared **data** with" instead of "Shared **credentials** with"?

Comment (daniel-mader) [reply unime/src/i18n/en/index.ts]: Same for "Received **data** from", right?

File: unime/src/lib/components/molecules/ListItemCard.svelte [hunk 29]
```diff
@@ -1,6 +1,8 @@
 <script lang="ts">
   import { createEventDispatcher } from 'svelte';
 
+  import { twMerge } from 'tailwind-merge';
```
Comment (daniel-mader) [inline unime/src/lib/components/molecules/ListItemCard.svelte]: Unused import

File: identity-wallet/src/state/reducers/history.rs
Comment (Oran-Dan) [inline identity-wallet/src/state/reducers/history.rs:1]: These structs are in fact not reducers, so they are in the wrong folder. Anticipating the folder restructure, you can create a shared folder and add them to a separate shared/history.rs file. This will be the required structure when your branch gets merged. It will still create a conflict but then you won't have to change the imports anymore after merging.

**Comment by Oran-Dan** [comment]
I suggest we use only one name for the dragon profile.
A bit confusing the Shenron and dragon double naming. 
Since all functions and enums use Dragon i suggest we just use Dragon 100% also as name and leave the Shenron.
Sorry for commenting this way too late on a wrong PR :|

File: unime/src/i18n/en/index.ts (hunk 28 above)
Comment (Norlock) [reply unime/src/i18n/en/index.ts]: Yes but I discussed with Andrew and he seems to prefer credentials as well, however if more than just credentials will be shared we can use data. Its more used as an umbrella term which could include more.

File: unime/src/i18n/en/index.ts [hunk 30]
```diff
@@ -207,7 +207,7 @@ const en = {
     NAVBAR_TITLE: 'Connected',
     TABS: {
       CONNECTIONS: 'Connections',
-      TIMELINE: 'Timeline',
+      HISTORY: 'History',
```
Comment (nanderstabel) [inline unime/src/i18n/en/index.ts:210]: Can you update these tags for NL and DE as well? (Same for `HISTORY` below in line 255)

**Review by nanderstabel — DISMISSED** [review]

**Review by Oran-Dan — DISMISSED** [review]

File: unime/.prettierignore (hunk 14 above)
Comment (daniel-mader) [reply unime/.prettierignore]: The file `.tsconfig.json` does not exist. You might mean `tsconfig.json`, but I suggest you undo that ignore and reintroduce it if you encounter issues again.

**Review by nanderstabel — APPROVED** [review]

**Review by daniel-mader — APPROVED** [review]
